### PR TITLE
update pre-commit  and add pyupgrade

### PR DESCRIPTION
**Pull request by Czaki**
# Description

As there is a discussion about usage pre-commit.ci (which also upgrades hooks version) Maybe it is a good time to upgrade the version of the current hooks. 

Also maybe add `pyupgrade` which uses the best language construction for chosen object creation.

## Type of change

- [x] Refactor 

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by sofroniewn** [comment]
Exciting @Czaki can you run this again though with `exclude: _vendor|vendored|examples` see
https://github.com/napari/napari/blob/e274ab842041e45dca6da133f696c5e0c18ad2c5/.pre-commit-config.yaml#L12

We might want to include `examples` at some point. But definitely want to still exclude `_vendor|vendored`

File: napari/_qt/layer_controls/qt_points_controls.py [hunk 1]
```diff
@@ -187,7 +187,7 @@ def mouseMoveEvent(self, event):
         self.layer.status = self.layer.mode
 
     def _on_mode_change(self, event):
-        """"Update ticks in checkbox widgets when points layer mode is changed.
+        """ "Update ticks in checkbox widgets when points layer mode is changed.
```
Comment (sofroniewn) [inline napari/_qt/layer_controls/qt_points_controls.py]: manual fix

File: napari/_qt/layer_controls/qt_shapes_controls.py [hunk 2]
```diff
@@ -236,7 +236,7 @@ def mouseMoveEvent(self, event):
         self.layer.status = str(self.layer.mode)
 
     def _on_mode_change(self, event):
-        """"Update ticks in checkbox widgets when shapes layer mode changed.
+        """ "Update ticks in checkbox widgets when shapes layer mode changed.
```
Comment (sofroniewn) [inline napari/_qt/layer_controls/qt_shapes_controls.py]: manual fix

File: napari/_qt/layer_controls/qt_vectors_controls.py [hunk 3]
```diff
@@ -235,7 +235,7 @@ def _on_length_change(self, event=None):
             self.lengthSpinBox.setValue(self.layer.length)
 
     def _on_edge_width_change(self, event=None):
-        """"Receive layer model width change event and update width spinbox.
+        """ "Receive layer model width change event and update width spinbox.
```
Comment (sofroniewn) [inline napari/_qt/layer_controls/qt_vectors_controls.py]: manual fix

File: napari/_qt/layer_controls/qt_vectors_controls.py [hunk 4]
```diff
@@ -246,7 +246,7 @@ def _on_edge_width_change(self, event=None):
             self.widthSpinBox.setValue(self.layer.edge_width)
 
     def _on_edge_color_mode_change(self, event=None):
-        """"Receive layer model edge color mode change event & update dropdown.
+        """ "Receive layer model edge color mode change event & update dropdown.
```
Comment (sofroniewn) [inline napari/_qt/layer_controls/qt_vectors_controls.py]: manual fix

File: napari/_qt/layer_controls/qt_vectors_controls.py [hunk 5]
```diff
@@ -263,7 +263,7 @@ def _on_edge_color_mode_change(self, event=None):
             self._update_edge_color_gui(mode)
 
     def _on_edge_color_change(self, event=None):
-        """"Receive layer model edge color  change event & update dropdown.
+        """ "Receive layer model edge color  change event & update dropdown.
```
Comment (sofroniewn) [inline napari/_qt/layer_controls/qt_vectors_controls.py]: manual fix

File: napari/_qt/layer_controls/qt_points_controls.py (hunk 1 above)
Comment (Czaki) [reply napari/_qt/layer_controls/qt_points_controls.py]: it means remove additional `"`?

**Review by sofroniewn — COMMENTED** [review]
Looking good. I flagged some times when we had `""""` instead of `"""` that need to be manually fixed, but otherwise looks good

File: napari/_qt/layer_controls/qt_points_controls.py (hunk 1 above)
Comment (sofroniewn) [reply napari/_qt/layer_controls/qt_points_controls.py]: yeah, this should just be 3 of the `"` and get rid of the extra one at beginning/ end. was a typo initially

**Comment by Czaki** [comment]
This change may need the information to PR developers, to copy only `.pre-commit-config.yaml`, not merge the whole `master` branch.

**Comment by sofroniewn** [comment]
> This change may need the information to PR developers, to copy only .pre-commit-config.yaml, not merge the whole master branch.

you mean for people with open existing PRs?

**Comment by Czaki** [comment]
> you mean for people with open existing PRs?

Yes

**Comment by tlambert03** [comment]
pyupgrade is interesting, but I'd kinda like to be able to evaluate it on it's own ... but would it be possible to do this in two PRs, so we can see how much of this is due to the changes in black 20.8b1, and how much is due to pyupgrade?

**Comment by Czaki** [comment]
> pyupgrade is interesting, but I'd kinda like to be able to evaluate it on it's own ... but would it be possible to do this in two PRs, so we can see how much of this is due to the changes in black 20.8b1, and how much is due to pyupgrade?

There are two separates commits. First with black and second with pyupgrade. But I could split.

**Comment by tlambert03** [comment]
> There are two separates commits. First with black and second with pyupgrade. But I could split.

ahh thanks!  I missed that.

**Comment by tlambert03** [comment]
oh i like the pyupgrade changes!

**Review by tlambert03 — APPROVED** [review]

**Review by tlambert03 — COMMENTED** [review]
actually, could you also update flake8 and black in setup.cfg as well?  I know they're not necessary for pre-commit, which runs in its own env... but it is helpful to be running the same versions in your IDE linter and stuff

**Comment by Czaki** [comment]
why `check-manifest` is in `dev` dependencies and not in pre-commit?

File: setup.cfg [hunk 6]
```diff
@@ -101,9 +101,9 @@ release =
     pygithub
     tqdm
 dev = 
-    pre-commit>=1.16.1
-    black==19.10b0
-    flake8==3.8.3
+    pre-commit>=2.9.0
```
Comment (Czaki) [inline setup.cfg:104]: `isort` requirement https://github.com/PyCQA/isort/blob/develop/.pre-commit-hooks.yaml

**Comment by tlambert03** [comment]
> why `check-manifest` is in `dev` dependencies and not in pre-commit? 

Cause check manifest is rather slow to run, so it's annoying to have it run on every commit.

**Comment by sofroniewn** [comment]
Ok, I think this is now good to go! Thanks for those updates @Czaki I'll merge promptly

**Review by sofroniewn — APPROVED** [review]

**Pull request merged**